src/sinks/util/auto_concurrency/controller.rs

```diff
@@ -229,8 +229,14 @@ where
         start: Instant,
         response: &Result<L::Response, crate::Error>,
     ) {
-        let is_back_pressure = match response {
-            Ok(_) => false,
+        // It would be better to avoid generating the string in Retry(_)
+        // just to throw it away here, but it's probably not worth the
+        // effort.
+        let response = response
```

Comment:
Should this variable be called something like `retry_status`? It doesn't seem to represent the `response` anymore after being mapped which confused me initially.

Reply:
Sure, I'll rename.

Reply:
> This looks good! Are there tests for this behavior?

Not yet. I'll see about adding that to #3802 

> I reran the concurrency tests and it seems to be handling 429s and 503s well now.

Yay! One step closer. I will say the throughput graphs look a bit ugly, but eyeballing the average looks like it's about right. It would be good to get some numbers on the average for those two, as they still have some surprisingly deep dips.